lib/usd/translators/jointWriter.cpp

```diff
@@ -167,27 +167,22 @@ static bool _FindDagPoseMembers(
     std::vector<unsigned int>*   indices)
 {
     MStatus status;
-    MPlug   membersPlug = dagPoseDep.findPlug("members", &status);
+    MPlug   membersPlug = dagPoseDep.findPlug("members", false, &status);
     CHECK_MSTATUS_AND_RETURN(status, false);
 
     // Build a map of dagPath->index.
-
-    struct _HashObjectHandle
-    {
-        std::size_t operator()(const MObjectHandle& o) const { return o.hashCode(); }
-    };
-
-    std::unordered_map<MObjectHandle, size_t, _HashObjectHandle> pathIndexMap;
+    UsdMayaUtil::MObjectHandleUnorderedMap<size_t> pathIndexMap;
     for (size_t i = 0; i < dagPaths.size(); ++i) {
         pathIndexMap[MObjectHandle(dagPaths[i].node())] = i;
     }
 
     MPlugArray inputs;
 
+    size_t numDagPaths = dagPaths.size();
     indices->clear();
-    indices->resize(
-        std::min(membersPlug.numElements(), static_cast<unsigned int>(dagPaths.size())), -1);
+    indices->resize(numDagPaths);
 
+    std::vector<uint8_t> visitedIndices(numDagPaths, 0);
```

Comment:
I would think `std::vector<bool>` here would be more compact?

Reply:
(Hi! I'm the engineer that's working on this PR and previously implemented our hotfix internally, and making @dgovil suffer being the middleman :P)

I think I generally avoid `std::vector<bool>` as the guarantees about compactness and contiguous memory storage vary across implementation, along with a host of other issues (i.e. http://www.gotw.ca/publications/N1211.pdf and https://isocpp.org/blog/2012/11/on-vectorbool). It's not a complete dealbreaker, but I'd argue that an array of ints that is guaranteed to be stored contiguously without worrying about future issues is easier than saving on bytes here since we're generally going to be iterating over the entire array in the common case. But if there's a strong opinion that we should squeeze out every last bit we can, again, not a dealbreaker; just let me know and I can swap it out.

Reply:
Hey @ysiewappl! Thanks for the info! I suspected that might be the case, but just wasn't sure without any additional context. No strong opinions in favor of `std::vector<bool>`, so this is fine by me!